.github/workflows/test_and_release.yml

```diff
@@ -0,0 +1,117 @@
+name: Test and Release
+
+on:
+  workflow_dispatch:
+  release:
+    types: [published]
+  pull_request:
+    branches:
+      - master
+      - v[0-9]+.[0-9]+-dev
+
+jobs:
+  test:
+    name: Run wallet-lib tests
+    runs-on: ubuntu-20.04
+    timeout-minutes: 60
```

Comment:
```suggestion
    timeout-minutes: 15
```

Reply:
This isn't always going to be enough if cache misses, the previous run took 17m

Reply:
Oh... You right. Let's keep it.